pkg/smartcontract/parameter.go

```diff
@@ -362,6 +362,22 @@ func NewParameterFromValue(value any) (Parameter, error) {
 			}
 			result.Type = ArrayType
 			result.Value = res
+		case reflect.Map:
+			res := make([]ParameterPair, 0, rv.Len())
+			iter := rv.MapRange()
+			for iter.Next() {
+				k, err := NewParameterFromValue(iter.Key().Interface())
+				if err != nil {
+					return result, err
+				}
+				v, err := NewParameterFromValue(iter.Value().Interface())
+				if err != nil {
+					return result, err
+				}
```

Comment:
It would be nice to wrap these errors with additional details, at least to mention is it a `key` or a `value` that is invalid. It will help to debug invalid conversion cases, it's mostly for users.

Reply:
Ditto for array, index of invalid item will be helpful.